api/v1beta1/foundationdbcluster_types.go

```diff
@@ -719,13 +728,45 @@ func (cluster *FoundationDBCluster) CheckReconciliation() (bool, error) {
 	}
 
 	if reconciled {
-		cluster.Status.Generations = GenerationStatus{
+		cluster.Status.Generations = ClusterGenerationStatus{
 			Reconciled: cluster.ObjectMeta.Generation,
 		}
 	}
 	return reconciled, nil
 }
 
+// GetDesiredAgentCount determines how many backup agents we should run
+// for a cluster.
+func (backup *FoundationDBBackup) GetDesiredAgentCount() int {
+	desiredAgentCount := backup.Spec.AgentCount
+	if desiredAgentCount < 0 {
+		return 0
+	} else if desiredAgentCount == 0 {
+		return 2
```

Comment:
If the desiredAgentCount is zero in the spec, that's equivalent to unset, so we're using 2 as a default.

We do something similar with the process counts for the main fdbserver processes. I agree that it's confusing, and I've been wondering if we should change our approach. Many of the built-in controllers, including Deployment, make the replica count a pointer to an integer rather than an integer so that they can distinguish between "set to zero" and "not set". Maybe we should do the same. It's a bit of a pain to work with in its own way, but it may be more clear.

Reply:
Yeah, I think it's really confusing, especially since 0 is a legitimate number of agents to have (looks like that'd be chosen by using -1). I'm fine if we push this through and I can file an issue to change the default values to not use 0 as unset, does that work?

Reply:
If we're going to do it, it's probably better to do it here rather than have to make a breaking change or add a new field and deprecate the old one.